Sources/MapboxMaps/Foundation/Camera/CameraManager.swift

```diff
@@ -174,122 +173,109 @@ public class CameraManager {
+        flyToAnimator.addCompletion { [weak self](position) in
+            // Call the developer-provided completion (if present)
+            completion?(position)
+            self?.internalAnimator = nil
         }
 
-        animator.startAnimation()
+        flyToAnimator.startAnimation()
+        internalAnimator = flyToAnimator
 
-        return animator
+        return internalAnimator
     }
 
-    /// This function optimizes the bearing for set camera so that it is taking the shortest path.
+    /// Ease the camera to a destination
     /// - Parameters:
-    ///   - startBearing: The current or start bearing of the map viewport.
-    ///   - endBearing: The bearing of where the map viewport should end at.
-    /// - Returns: A `CLLocationDirection` that represents the correct final bearing accounting for positive and negatives.
-    internal func optimizeBearing(startBearing: CLLocationDirection?, endBearing: CLLocationDirection?) -> CLLocationDirection? {
-        // This modulus is required to account for larger values
-        guard
-            let startBearing = startBearing?.truncatingRemainder(dividingBy: 360.0),
-            let endBearing = endBearing?.truncatingRemainder(dividingBy: 360.0)
-        else {
-            return nil
+    ///   - camera: the target camera after animation
+    ///   - duration: duration of the animation
+    ///   - completion: completion to be called after animation
+    /// - Returns: An instance of `CameraAnimatorProtocol` which can be interrupted if necessary
+    @discardableResult
+    public func ease(to camera: CameraOptions, duration: TimeInterval, completion: AnimationCompletion? = nil) -> CameraAnimatorProtocol? {
+
+        internalAnimator?.stopAnimation()
+
+        let animator = makeAnimator(duration: duration, curve: .easeInOut) { (transition) in
+            transition.center.toValue = camera.center
+            transition.padding.toValue = camera.padding
+            transition.anchor.toValue = camera.anchor
+            transition.zoom.toValue = camera.zoom
+            transition.bearing.toValue = camera.bearing
+            transition.pitch.toValue = camera.pitch
         }
 
-        // 180 degrees is the max the map should rotate, therefore if the difference between the end and start point is
-        // more than 180 we need to go the opposite direction
-        if endBearing - startBearing >= 180 {
-            return endBearing - 360
+        // Nil out the `internalAnimator` once the "ease to" finishes
+        animator.addCompletion { [weak self] (_) in
+            self?.internalAnimator = nil
         }
 
-        // This is the inverse of the above, accounting for negative bearings
-        if endBearing - startBearing <= -180 {
-            return endBearing + 360
+        // Add the developer-provided completion (if present)
+        if let completion = completion {
+            animator.addCompletion(completion)
         }
```

Comment:
seems simpler to just `completion?(position)` inside the completion block above

Reply:
This is done.